include/psa/crypto_types.h

```diff
@@ -380,4 +380,54 @@ typedef uint16_t psa_key_derivation_step_t;
 
 /**@}*/
 
+/** \defgroup pake Password-authenticated key exchange (PAKE)
+ * @{
+ */
+
+/** \brief Encoding of the side of PAKE
+ *
+ * Encodes which side of the protocol is being executed. For more information
+ * see the documentation of individual PSA_PAKE_SIDE_XXX constants.
+ */
+typedef uint16_t psa_pake_side_t;
+
+/** Encoding of input and output indicators for PAKE.
+ *
+ * Some PAKE protocols need to exchange more data than just a single key share.
+ * This type is for encoding additional input and output data for such
+ * protocols.
+ */
+typedef uint8_t psa_pake_data_t;
+
+/** Encoding of the type of the PAKE's primitive.
+ *
+ * Values defined by this standard will never be in the range 0x80-0xff.
+ * Vendors who define additional types must use an encoding in this range.
+ */
+typedef uint8_t psa_pake_primitive_type_t;
+
+/** Encoding of the bitsize for the PAKE's primitive.
+ *
+ * The type and family is not enough to identify the primitive to use in the
+ * PAKE, the implementation needs to know the bitsize too.
+ */
+typedef uint16_t psa_pake_bits_t;
```

Comment:
Since it is declared as being a mirror of `psa_key_bits_t`, why not typedef it to that? Or even just use `psa_key_bits_t` straight-up?

Reply:
I feel uncomfortable about using `psa_key_bits_t` directly, as I mentioned below I think there is value in separating these types, because as far as the PSA Crypto API is concerned, this information is not related to keys:
https://github.com/ARMmbed/mbedtls/pull/4317#discussion_r614098441

Using typedef ties these two types together, which seems to make sense since in the case of primitives involved with J-PAKE the semantics of these two types is tied together. However this is not true in general and later the types might diverge.

Also, the macros handling them are different and if we use typedef, changing `psa_key_bits_t` might cause bugs in the macros using `psa_pake_bits_t` that would be an increased chance of being missed during review.

Reply:
`psa_key_bits_t` is not a type defined by PSA Crypto (even if it does exist in mbedtls). The PSA Cyprto API uses `size_t` for key bit sizes.

If the use of this in pake is to identify a specific family member, then it _is_ a mirror for how the bit size is used in an asymmetric key. For asymmetric keys, the bit-size does not identify the literal size of the key, it specifies the specific member of the key family that is defined by the key type.